packages/zoe/src/contractSupport/ratio.js

```diff
@@ -391,7 +369,9 @@ export const parseRatio = (
   denominatorBrand = numeratorBrand,
 ) => {
   const match = `${numeric}`.match(NUMERIC_RE);
-  assert(match, X`Invalid numeric data: ${numeric}`);
+  if (!match) {
+    throw Fail`Invalid numeric data: ${numeric}`;
+  }
 
   const [whole, part = ''] = [match[1], match[2]];
```

Comment:
Only arguably in scope for this PR, but there are faster approaches here too.
```suggestion
  const whole = match[1];
  const part = match[2] || '';
```
```sh
#### Moddable XS
destructure wrapping array: 0.94 ops/ms
destructure match: 1.10 ops/ms
destructure match.slice: 0.92 ops/ms
no destructure: 1.70 ops/ms

#### V8
destructure wrapping array: 16.13 ops/ms
destructure match: 16.39 ops/ms
destructure match.slice: 12.35 ops/ms
no destructure: 17.54 ops/ms
```

Reply:
Unless it makes an overall difference, I prefer the clarity of the destructuring. In this case, I would be surprised if it made an overall difference. Resolving.